include/mirage/search/search.h

```diff
@@ -89,46 +78,52 @@ class KernelGraphGenerator {
 
   char const *filename;
   std::vector<json> generated_graphs;
+  int num_thread;
+
+  std::chrono::milliseconds timeout;
 
 private:
   std::vector<std::shared_ptr<AlgebraicPattern>> final_patterns;
-  std::unordered_map<DTensor, std::shared_ptr<AlgebraicPattern>>
+  std::unordered_map<int, std::shared_ptr<AlgebraicPattern>>
```

Comment:
Is this still a map from dtensor to AlgebraicPattern? If so, shouldn't the key be int64_t?

Reply:
It should be int64_t. Pushed the updates.